### PR TITLE
Copy ParseCgroupFileUnified and Drop rest of containerd/cgroups

### PR DESCRIPTION
**Pull request by dims**
Related to discussion in https://github.com/kubernetes/kubernetes/issues/128157 - Let's ensure we use ONLY one cgroups library. 

Here in this PR we can keep a temporary copy until we get a longer term solution in `runc`'s cgroup library, see https://github.com/opencontainers/runc/pull/4497

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: third_party/forked/cgroups/parse_utils.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+   Copyright The containerd Authors.
```
Comment (dims) [inline third_party/forked/cgroups/parse_utils.go:2]: copied the entire file, so keeping their copyright

File: third_party/forked/cgroups/parse.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2024 The Kubernetes Authors.
```
Comment (dims) [inline third_party/forked/cgroups/parse.go:2]: copied just a couple of methods into a new file, so just adding ours.

**Comment by haircommander** [comment]
/lgtm
/hold

thanks for picking this up! holding to let CI run and anyone else chime in

**Comment by dims** [comment]
Weird that i can't repro the failure locally

```
+++ working dir: /home/prow/go/src/k8s.io/kubernetes
+++ command: bash "hack/verify-testing-import.sh"
Testing package imported in:
	/home/prow/go/src/k8s.io/kubernetes/cmd/kubelet
+++ exit code: 1
+++ error: 1
```

**Comment by dims** [comment]
> could it be a mac vs linux issue? digging deeper

yep. i see the failure on linux... will think about how to do this better in my long TODO later :)

for now, i had the wrong file name for the tests!

**Pull request merged**